pkg/datarepair/checker/checker.go

```diff
@@ -19,3 +30,107 @@ func (c *Config) Run(ctx context.Context) (err error) {
 
 	return err
 }
+
+
+// Checker contains the information needed to do checks for missing pieces
+type Checker struct {
+	params *pb.IdentifyRequest
+	pointerdb storage.KeyValueStore
+	repairQueue *queue.Queue
+	overlay pb.OverlayServer
+	logger *zap.Logger
+}
+
+var (
+	mon          = monkit.Package()
+	checkerError = errs.Class("data repair checker error")
+	
+)
+
+// NewChecker creates a new instance of checker
+func NewChecker(params *pb.IdentifyRequest, pointerdb storage.KeyValueStore, repairQueue *queue.Queue, overlay pb.OverlayServer, logger *zap.Logger) *Checker {
+	return &Checker{
+		params: params,
+		pointerdb: pointerdb,
+		repairQueue: repairQueue,
+		overlay: overlay,
+		logger: logger,
+	}
+}
+
+// IdentifyInjuredSegments checks for missing pieces off of the pointerdb and overlay cache
+func (c *Checker) IdentifyInjuredSegments(ctx context.Context) (err error) {
+	defer mon.Task()(&ctx)(&err)
+	c.logger.Debug("entering pointerdb iterate")
+	err = c.pointerdb.Iterate(storage.IterateOptions{Prefix: storage.Key(c.params.Prefix), First: storage.Key(c.params.First), Recurse: c.params.Recurse, Reverse: c.params.Reverse},
+		func(it storage.Iterator) error {
+			var item storage.ListItem
+			if c.params.Limit <= 0 || c.params.Limit > storage.LookupLimit {
+				c.params.Limit = storage.LookupLimit
+			}
+			for ; c.params.Limit > 0 && it.Next(&item); c.params.Limit-- {
+				pointer := &pb.Pointer{}
+				err = proto.Unmarshal(item.Value, pointer)
+				if err != nil {
+					return checkerError.New("error unmarshalling pointer %s", err)
+				}
+				pieces := pointer.Remote.RemotePieces
+				var nodeIDs []dht.NodeID
+				for _, p := range pieces {
+					nodeIDs = append(nodeIDs, kademlia.StringToNodeID(p.NodeId))
+				}
+				missingPieces, healthyPieces, err := c.offlineAndOnlineNodes(ctx, nodeIDs)
```

Comment:
`offlinePiece, onlinePieces`... We don't know if they are healthy until we try to download them. Data can get corrupt.

Reply:
That is a good point. However, there was discussion with @jtolds  and @kaloyan-raev that we could take that risk?

Reply:
I just mean the naming... yes, we kind of consider them "healthy" unless proven otherwise. Just trying to keep consistency with the method name.